sdk/iot/core/src/az_iot_telemetry.c

```diff
@@ -0,0 +1,95 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+#include <stdint.h>
+
+#include "az_iot_hub_client.h"
+#include <az_contract_internal.h>
+#include <az_result.h>
+#include <az_span.h>
+
+static const uint8_t telemetry_prop_delim = '?';
+static const uint8_t telemetry_prop_separator = '&';
+static const uint8_t telemetry_null_terminator = 0;
+static const az_span telemetry_topic_prefix = AZ_SPAN_LITERAL_FROM_STR("devices/");
+static const az_span telemetry_topic_modules_mid = AZ_SPAN_LITERAL_FROM_STR("/modules/");
+static const az_span telemetry_topic_suffix = AZ_SPAN_LITERAL_FROM_STR("/messages/events/");
+
+az_result az_iot_hub_client_telemetry_publish_topic_get(
+    az_iot_hub_client const* client,
+    az_iot_hub_client_properties const* properties,
+    az_span mqtt_topic,
+    az_span* out_mqtt_topic)
+{
+  AZ_CONTRACT_ARG_NOT_NULL(client);
+  AZ_CONTRACT_ARG_VALID_SPAN(mqtt_topic);
+  AZ_CONTRACT_ARG_NOT_NULL(out_mqtt_topic);
+
+  az_result result;
+
+  int32_t required_size = 0;
```

Comment:
why set to 0 if the next line populates it?

Reply:
it was using += before so I had to set it. Will remove